Classes/YLRefreshHeaderView/YLRefreshHeaderView.m

```diff
@@ -31,26 +35,37 @@ - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState {
 }
 
 - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState animated:(BOOL)animated {
-  _refreshState = refreshState;
+  if (_refreshState != refreshState) {
+    _refreshState = refreshState;
 
-  void (^animationBlock)() = ^{
-    UIScrollView *strongScrollView = self.scrollView;
-    UIEdgeInsets contentInset = strongScrollView.contentInset;
-    // TODO(mglidden): make this support view controllers that have UIRectEdgeTop
-    contentInset.top = (refreshState == YLRefreshHeaderViewStateRefreshing) ? self.pullAmountToRefresh : 0;
-    strongScrollView.contentInset = contentInset;
-  };
-  void (^completionBlock)(BOOL) = ^(BOOL finished) {
-    if (self.refreshState == YLRefreshHeaderViewStateClosing) {
-      self.refreshState = YLRefreshHeaderViewStateNormal;
+    void (^animationBlock)() = ^{
+      UIScrollView *strongScrollView = self.scrollView;
+      UIEdgeInsets contentInset = strongScrollView.contentInset;
+      switch (refreshState) {
+        case YLRefreshHeaderViewStateRefreshing:
+          self.previousTopInset = contentInset.top;
+          contentInset.top = self.pullAmountToRefresh;
```

Comment:
Blake, I don't understand this response. What about the example
implementation breaking? The bottom line is that the behavior is not the
correct one given the example. Can you fix it and still keep that example
code from breaking? Are you asking for help to find the solution?

--Jimmy Ho
iOS Engineer | Reader Experience Team


On Tue, Aug 2, 2016 at 1:38 PM, Blake Tsuzaki <notifications@github.com>
wrote:

> In Classes/YLRefreshHeaderView/YLRefreshHeaderView.m
> <https://github.com/Yelp/YLTableView/pull/25#discussion_r73232408>:
>
> > -    UIScrollView *strongScrollView = self.scrollView;
> > -    UIEdgeInsets contentInset = strongScrollView.contentInset;
> > -    // TODO(mglidden): make this support view controllers that have UIRectEdgeTop
> > -    contentInset.top = (refreshState == YLRefreshHeaderViewStateRefreshing) ? self.pullAmountToRefresh : 0;
> > -    strongScrollView.contentInset = contentInset;
> > -  };
> > -  void (^completionBlock)(BOOL) = ^(BOOL finished) {
> > -    if (self.refreshState == YLRefreshHeaderViewStateClosing) {
> > -      self.refreshState = YLRefreshHeaderViewStateNormal;
> > +    void (^animationBlock)() = ^{
> > +      UIScrollView *strongScrollView = self.scrollView;
> > +      UIEdgeInsets contentInset = strongScrollView.contentInset;
> > +      switch (refreshState) {
> > +        case YLRefreshHeaderViewStateRefreshing:
> > +          self.previousTopInset = contentInset.top;
> > +          contentInset.top = self.pullAmountToRefresh;
>
> Ah, I'm guessing the 64px is the height of the navigation bar overlapping
> the tableview. I'm not sure if this is something I should change, because
> the previous implementation assumed that contentInset.top is initially
> zero, resets to zero, and that the top edge of the table view is aligned
> with the bottom edge of the navigation bar. So if you have a
> pullAmountToRefresh of 15, the new contentInset.top would be 15 as well.
> Implementations that use YLTableView may already rely on this behavior.
> This would present a bigger challenge when contentInset.top is initially
> negative, which would mean part of the refresh header would be cut off
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Yelp/YLTableView/pull/25/files/3888f96922642f04405616f03a20639ec9d94743#r73232408>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIbjIpOg6gNkKbVavDV2NTqDxtTy7yYDks5qb6q0gaJpZM4JXr-B>
> .
>